src/Uno.UWP/Storage/StorageFolder.cs

```diff
@@ -176,42 +176,45 @@ internal StorageFolder(string name, string path)
 
 		public IAsyncOperation<StorageFile> CreateFileAsync(string desiredName) => CreateFileAsync(desiredName, CreationCollisionOption.FailIfExists);
 
-        public IAsyncOperation<StorageFile> CreateFileAsync(string path, CreationCollisionOption option) =>
+        public IAsyncOperation<StorageFile> CreateFileAsync(string desiredName, CreationCollisionOption option) =>
             AsyncOperation.FromTask(async ct =>
             {
-                if (File.Exists(global::System.IO.Path.Combine(Path, path)))
+                if (File.Exists(global::System.IO.Path.Combine(Path, desiredName)))
                 {
                     switch (option)
                     {
+						case CreationCollisionOption.FailIfExists:
+							throw new Exception("Cannot create a file when that file already exists.");
```

Comment:
I agree with the robot, but it does not matter much if the exception type is not the same as UWP here. We can use a more acceptable exception type, like `InvalidOperation` for instance.

Reply:
But if someone creates code with catch, and implemented logic for handling this exception, then changing exception that is thrown (another in UWP, another in Uno) breaks such code. I think Uno should be as far UWP as even "bug compatible" - just to not break user code that expects some behaviour, even expects known UWP errors.